sound/soc/sof/intel/hda.c

```diff
@@ -572,7 +572,8 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	sdev->bar[HDA_DSP_BAR] = pci_ioremap_bar(pci, HDA_DSP_BAR);
 	if (!sdev->bar[HDA_DSP_BAR]) {
 		dev_err(&pci->dev, "error: ioremap error\n");
-		return -ENXIO;
+		ret = -ENXIO;
+		goto hdac_bus_unmap;
 	}
```

Comment:
This 'goto hdac_bus_unmap' is fine, but we may need add some error point e.g. 'dsp_bar_unmap' where we will do iounmap for sdev->bar[HDA_DSP_BAR].